File: src/main/kotlin/dev/jefy/connectpro/recommandation/domain/vo/EventTrackingId.java
package dev.jefy.connectpro.recommandation.domain.vo;


import org.springframework.util.Assert;

import java.util.UUID;

import dev.jefy.connectpro.shared.infrastructure.ddd.DIdentifier;

/**
 * @author Jôph Yamba
 */
public record EventTrackingId (UUID value) implements DIdentifier<UUID> {
    public EventTrackingId {
        Assert.notNull(value, "value cannot be null");
    }
    public static EventTrackingId generate() {
        return new EventTrackingId(UUID.randomUUID());
    }
}

File: src/main/kotlin/dev/jefy/connectpro/recommandation/domain/vo/EventTrackingId.kt
package dev.jefy.connectpro.recommandation.domain.vo

import jakarta.persistence.Embeddable
import java.util.*

@Embeddable
data class EventTrackingId(var value: UUID)  {
    companion object {
        fun generate(): EventTrackingId = EventTrackingId(UUID.randomUUID())
        fun of(value: UUID): EventTrackingId = EventTrackingId(value)
    }
}